econml/tests/test_dowhy.py

```diff
@@ -0,0 +1,66 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+import numpy as np
+import unittest
+from econml.dml import LinearDML, CausalForestDML
+from econml.orf import DROrthoForest
+from econml.dr import DRLearner
+from econml.metalearners import XLearner
+from econml.iv.dml import DMLATEIV
+from sklearn.linear_model import LinearRegression, LogisticRegression, Lasso
+
+
+class TestDowhy(unittest.TestCase):
+
+    def _get_data(self):
+        X = np.random.normal(0, 1, size=(500, 5))
+        T = np.random.binomial(1, .5, size=(500,))
+        Y = np.random.normal(0, 1, size=(500,))
+        Z = np.random.normal(0, 1, size=(500,))
+        return Y, T, X[:, [0]], X[:, 1:], Z
+
+    def test_dowhy(self):
+        def reg():
+            return LinearRegression()
+
+        def clf():
+            return LogisticRegression()
+
+        Y, T, X, W, Z = self._get_data()
+        # test at least one estimator from each category
+        models = {"dml": LinearDML(model_y=reg(), model_t=clf(), discrete_treatment=True,
+                                   linear_first_stages=False),
+                  "dr": DRLearner(model_propensity=clf(), model_regression=reg(),
+                                  model_final=reg()),
+                  "xlearner": XLearner(models=reg(), cate_models=reg(), propensity_model=clf()),
+                  "cfdml": CausalForestDML(model_y=reg(), model_t=clf(), discrete_treatment=True),
+                  "orf": DROrthoForest(n_trees=10, propensity_model=clf(), model_Y=reg()),
+                  "dmlateiv": DMLATEIV(model_Y_W=reg(),
+                                       model_T_W=clf(),
+                                       model_Z_W=reg(),
+                                       discrete_treatment=True,
+                                       discrete_instrument=False)}
+        for name, model in models.items():
+            with self.subTest(name=name):
+                est = model
+                if name == "xlearner":
+                    est_dowhy = est.dowhy.fit(Y, T, X=np.hstack((X, W)), W=None)
```

Comment:
Why do we have to stack X and W in this case?

Reply:
it's a personal choice, since metalearner we don't have W, I could either input X,W, or stack of them as X.